lib/puma/client.rb

```diff
@@ -85,6 +85,12 @@ def initialize(io, env=nil)
 
     def_delegators :@io, :closed?
 
+    # Test to see if io meets a bare minimum of functioning, @to_io needs to be
+    # used for MiniSSL::Socket
+    def io_ok?
+      ::BasicSocket === @to_io && !closed?
```

Comment:
personally I'd prefer `@to_io.is_a?(BasicSocket)`

Reply:
Sure.  I'm rebasing & since this should probably have a History.md entry, I'll update it.  I'll do the links later, as it's now two command lines...